lexicon/providers/yandexcloud.py

```diff
@@ -0,0 +1,307 @@
+
+    def _authenticate(self):
+        if self._get_provider_option("dns_zone_id"):
+            self.domain_id = self._get_provider_option("dns_zone_id")
+
+        cloud_id = self._get_cloud_id()
+        folder_id = self._get_folder_id(cloud_id)
+        self.domain_id = self._get_dns_zone_id(folder_id)
+
+    def _get_cloud_id(self) -> str:
+        """Gets Cloud ID from Resource Manager API
+        https://cloud.yandex.com/en/docs/resource-manager/api-ref/Cloud/list"""
+        if self._get_provider_option("cloud_id"):
+            return self._get_provider_option("cloud_id")
+
+        payload: Dict = self._get(f"{self.resources_api_endpoint}/clouds")
+
+        if "clouds" not in payload:
+            raise AuthenticationError("No clouds found")
+        if len(payload["clouds"]) > 1:
+            raise AuthenticationError("Too many clouds found, you must specify one explicitly, or set DNS zone ID")
+
+        return payload["clouds"][0]["id"]
+
+    def _get_folder_id(self, cloud_id: str) -> str:
+        """Gets Folder ID from Resource Manager API
+        https://cloud.yandex.com/en/docs/resource-manager/api-ref/Folder/list
+        """
+        if self._get_provider_option("folder_id"):
+            return self._get_provider_option("folder_id")
+
+        payload: Dict = self._get(f"{self.resources_api_endpoint}/folders", {"cloudId": cloud_id})
+
+        if "folders" not in payload:
+            raise AuthenticationError(f"No folders found for Cloud ID {cloud_id}")
+        if len(payload["folders"]) > 1:
+            raise AuthenticationError("Too many folders found, you must specify one explicitly, or set DNS zone ID")
+
+        return payload["folders"][0]["id"]
+
+    def _get_dns_zone_id(self, folder_id: str) -> str:
+        """Gets Domain ID from DnsZone API
+        https://cloud.yandex.com/en/docs/dns/api-ref/DnsZone/list
+        """
+        payload: Dict = self._get("/zones", {"folderId": folder_id, "filter": f"zone='{self.domain}.'"})
+
+        if "dnsZones" not in payload:
+            raise AuthenticationError(f"No domain found for Folder ID {folder_id}")
+        if len(payload["dnsZones"]) > 1:
+            raise AuthenticationError("Too many domains found, this should not happen")
+
+        return payload["dnsZones"][0]["id"]
+
+    def _list_records(self, rtype=None, name=None, content=None) -> List[Dict]:
+        """List all records. Return an empty list if no records found.
+        type, name and content are used to filter records.
+        https://cloud.yandex.com/en/docs/dns/api-ref/DnsZone/listRecordSets
+        """
+        records: List[Dict] = []
+
+        query_params = {"filter": ""}
+        if rtype:
+            query_params["filter"] = f"type='{rtype}'"
+
+        # name should be in full form, transform it in case it's not
+        if name:
+            name = self._full_name(name) + "."
```

Comment:
I see that everytime you call `self._full_name`, you append a dot to it. Maybe you could override the `_full_name()` method itself for this provider so that it always return the FQDN?

Reply:
Unfortunately out of three calls once I need not the dot, so I would prefer the current way.